base/compiler/typeinfer.jl

```diff
@@ -348,7 +374,8 @@ function maybe_compress_codeinfo(interp::AbstractInterpreter, linfo::MethodInsta
 end
 
 function transform_result_for_cache(interp::AbstractInterpreter, linfo::MethodInstance,
-                                    valid_worlds::WorldRange, @nospecialize(inferred_result))
+                                    valid_worlds::WorldRange, result::InferenceResult)
+    inferred_result = result.src
```

Comment:
The reason for this change is that identity of `InferenceResult` can be used as a key for inference cache and so it is sometimes very useful for customizing the behavior of caching mechanism of the native compilation pipeline.